File: hw/chicaiza/u2/hw19Inheritance/FarmSystemPython/model/farm_animal.py
from datetime import date
from model.cage import Cage

class FarmAnimal:
    def __init__(self, id: int, breed: str, born_on: date, gender: str,
                 is_able_to_reproduce: bool, weight: float, cage: Cage):
        self.id = id
        self.breed = breed
        self.born_on = born_on
        self.gender = gender
        self.is_able_to_reproduce = is_able_to_reproduce
        self.weight = weight
        self.cage = cage

    def get_age_in_months(self):
        today = date.today()
        return (today.year - self.born_on.year) * 12 + (today.month - self.born_on.month)

    def __str__(self):
        return (f"FarmAnimal(id={self.id}, breed={self.breed}, bornOn={self.born_on}, "
                f"gender={self.gender}, ableToReproduce={self.is_able_to_reproduce}, "
                f"weight={self.weight}, cage={self.cage})")

File: hw/quina/u2/hw19Inheritance/FarmSystemJS/ec.edu.espe.FarmSystem.model/FarmAnimal.js
import { Cage } from './Cage.js';

export class FarmAnimal {
  constructor(id, breed, bornOn, gender, isAbleToReproduce, weight, cage /** @type {Cage} */) {
    this.id = id;
    this.breed = breed;
    this.bornOn = bornOn;            // Date
    this.gender = gender;
    this.isAbleToReproduce = isAbleToReproduce;
    this.weight = weight;
    this.cage = cage;
  }

  toString() {
    return `FarmAnimal(id=${this.id}, breed='${this.breed}', bornOn=${this.bornOn.toISOString().slice(0,10)}, ` +
           `gender='${this.gender}', isAbleToReproduce=${this.isAbleToReproduce}, weight=${this.weight}, cage=${this.cage})`;
  }
}
